src/main/java/gov/nasa/pds/tools/validate/CrossLabelFileAreaReferenceChecker.java

```diff
@@ -33,20 +34,24 @@ private static String resolve (String name, ValidationTarget target) throws URIS
    * @throws SAXException
    * @throws URISyntaxException
    */
-  public static boolean add (String name, ValidationTarget target) throws IOException, ParserConfigurationException, SAXException, URISyntaxException {
+  public static boolean add (String name, ValidationTarget target, boolean isObs)
+      throws IOException, ParserConfigurationException, SAXException, URISyntaxException {
     boolean success = false;
     DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
     Document xml = dbf.newDocumentBuilder().parse(target.getUrl().openStream());
-    DOMSource domSource = new DOMSource(xml);    
+    DOMSource domSource = new DOMSource(xml);
+    String full_name = resolve(name, target);
+    isObservational.put(full_name,
+        (isObservational.containsKey(full_name) ? isObservational.get(full_name) : false) | isObs);
```

Comment:
Using the bitwise OR `|` here instead of the logical OR `||` means Java will always evaluate the right-hand side. Maybe this is a premature optimization, but perhaps it could be rewritten with `isObs || (isObservational…` and let `isObs` being `true` skip the `containsKey`/`get` call completely?

Reply:
Yes, that would be a CPU time improvement. Changed.